extension/js/common/platform/catch.ts

```diff
@@ -243,16 +244,36 @@ export class Catch {
       }
       const { line, col } = Catch.getErrorLineAndCol(e);
       const msg = e instanceof Error ? e.message : String(e);
-      Catch.onErrorInternalHandler(`REJECTION: ${msg}`, window.location.href, line, col, e, true);
+      Catch.onErrorInternalHandler(`REJECTION: ${msg}`, location.href, line, col, e, true);
     }
   };
 
-  public static setHandledInterval = (cb: () => void | Promise<void>, ms: number): number => {
-    return window.setInterval(Catch.try(cb), ms); // error-handled: else setInterval will silently swallow errors
+  public static setHandledInterval = (cb: () => void | Promise<void>, ms: number): string => {
+    // Couldn't set chrome.alrams here because chrome.alrams becomes undefined in gmail.com content script
+    const alarmName = `interval_${new Date().getTime()}_${ms}`;
+    BrowserMsg.send.bg.createAlarmWithDelay({ alarmName, ms });
+    BrowserMsg.addListener('perform_timeout_action', async ({ alarmName: intervalAlarmName }: Bm.PerformTimeoutAction) => {
+      if (alarmName === intervalAlarmName) {
+        Catch.try(cb)();
+      }
+    });
+    return alarmName;
```

Comment:
@sosnovsky As you can see, I've implemented the `setInterval` function as shown above. I had to create an alarm on the service worker side and then receive an event for the `perform_action`. As you know, we cannot send a callback (function) using `chrome.runtime.sendMessage`.

However, this approach doesn't seem ideal for a few reasons:
1. The `BrowserMsg.addListener('perform_timeout_action')` part won't be triggered upon completion of the `Catch.setHandledInterval` call or when the receiving end doesn't exist.
2. The code becomes somewhat cluttered.

Of course, we need to implement logic to use `setInterval` and `setTimeout` for normal pages (not service workers). But for service workers, I'm not sure what the best solution would be to implement `setInterval` and `setTimeout`.

Do you have better suggestion for implementing this?

Reply:
How about this one? @sosnovsky

Reply:
Probably we can move `chrome.alarms` code to service worker?
As I remember, we don't need to set background timers from content script code

Reply:
I mean here we can use old code with `setInterval`, and background timers using `chrome.alarms` will be set only in service worker, as they're initiated there

Reply:
Yep!
I thought too much.
It should be much easier